### PR TITLE
fix(core): Ensuring maxItemsToProcess is respected

### PR DESCRIPTION
**Pull request by jeyrschabu**
- fix candidateCounter with respect to maxItemsToProcess

### CODE REVIEW

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt [hunk 1]
```diff
@@ -280,7 +280,7 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
       val maxItemsToProcess = Math.min(preProcessedCandidates.size, getMaxItemsProcessedPerCycle(workConfiguration))
 
       for (candidate in preProcessedCandidates) {
-        if (candidateCounter.get() > maxItemsToProcess) {
+        if (candidateCounter.get() >= maxItemsToProcess) {
```
Comment (jeyrschabu) [inline swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt]: Makes sure to exit the loop when the maxItemsToProcess is reached.

**Comment by jeyrschabu** [comment]
Future state, I am thinking about handling counting inside of a `CycleStatCounter` or something similar (naming is hard). This would do for now.

**Comment by jeyrschabu** [comment]
Fixing tests

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt (hunk 1 above)
Comment (emjburns) [reply swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt]: HA 😄

**Comment by emjburns** [comment]
@jeyrschabu what's your thinking behind handling the counting somewhere else?

**Comment by jeyrschabu** [comment]
@emjburns right now we have a few counters tracking candidates (mark|delete|excluded|notified). They are a bit scattered and easy to miss. I am thinking along the lines of consolidating these counters and make them easy to assert in test.

**Pull request merged**